monitoring/bin/create_elasticsearch_datasource.sh

```diff
@@ -0,0 +1,203 @@
+        tenantNS=$2
+        shift 2
+      else
+        log_error "A value for parameter [NAMESPACE] has not been provided." >&2
+        show_usage
+        exit 2
+      fi
+      ;;
+    -t|--tenant)
+      if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+        tenant=$2
+        shift 2
+      else
+        log_error "A value for parameter [TENANT] has not been provided." >&2
+        show_usage
+        exit 2
+      fi
+      ;;
+    -h|--help)
+      show_usage
+      exit
+      ;;
+    -*|--*=) # unsupported flags
+      log_error "Unsupported flag $1" >&2
+      show_usage
+      exit 1
+      ;;
+    *) # preserve positional arguments
+      POS_PARMS="$POS_PARMS $1"
+      shift
+      ;;
+  esac
+done
+
+# Convert namespace and tenant to all lower-case
+tenantNS=$(echo "$tenantNS"| tr '[:upper:]' '[:lower:]')
+tenant=$(echo "$tenant"| tr '[:upper:]' '[:lower:]')
+
+# Check for parameters - set with cluster or tenant
+if [ -z "$tenantNS" ] && [ -z "$tenant" ]; then
+   cluster="true"
+elif [ -n "$tenant" ]; then
+    nst="${tenantNS}_${tenant}"
+else
+   nst="$tenantNS"
+fi
+
+# Check to see if monitoring/Viya namespace provided exists and components have already been deployed
+if [ "$cluster" == "true" ]; then
+    log_info "Checking for Grafana pods in the $MON_NS namespace ..."
+    if [[ $(kubectl get pods -n $MON_NS -l app.kubernetes.io/instance=v4m-prometheus-operator -o custom-columns=:metadata.namespace --no-headers | uniq | wc -l) == 0 ]]; then
+        log_error "No monitoring components found in the [$MON_NS] namespace."
+        log_error "Monitoring needs to be deployed in this namespace in order to configure Elasticsearch with this script.";
+        exit 1
+    else
+        log_debug "Monitoring found in $MON_NS namespace.  Continuing."
+    fi
+else
+    log_info "Checking the [$tenantNS] namespace for monitoring deployment for the [$tenant] tenant ..."
+    if [[ $(kubectl get pods -n $tenantNS -l app.kubernetes.io/instance=v4m-grafana-$tenant -o custom-columns=:metadata.namespace --no-headers | uniq | wc -l) == 0 ]]; then
+        log_error "No monitoring components were found for $tenant in the $tenantNS namespace."
+        log_error "Monitoring needs to be deployed in this namespace in order to configure Elasticsearch with this script.";
+        exit 1
+    else
+        log_debug "Monitoring deployment was found for the $tenant tenant in the $tenantNS namespace.  Continuing."
+    fi
+fi
+
+# Check to see if logging namespace provided exists and components have already been deployed
+log_info "Checking for Elasticsearch pods in the $LOG_NS namespace ..."
+if [[ $(kubectl get pods -n $LOG_NS -l app=v4m-es -o custom-columns=:metadata.namespace --no-headers | uniq | wc -l) == 0 ]]; then
+  log_error "No logging components found in the $LOG_NS namespace."
+  log_error "Logging needs to be deployed in this namespace in order to configure Elasticsearch with this script.";
```

Comment:
Reword messages.  The code checks for Elasticsearch, so the message should probably say it was not found.  And, this script does NOT configure _Elasticsearch_, it configures a(n Elasticsearch) datasource in Grafana.  

Perhaps:
`log_error "Elasticsearch was not found in the $LOG_NS namespace."`
`log_error "All of the required log monitoring components need to be deployed in this namespace before this script can configure the requested Grafana datasource "`